src/main/java/ladder/domain/line/LadderLineBuilder.java

```diff
@@ -0,0 +1,103 @@
+package ladder.domain.line;
+
+import ladder.domain.point.Direction;
+import ladder.domain.point.LadderPoint;
+import ladder.exception.OutOfSizeException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class LadderLineBuilder {
+    private final List<Direction> directions;
+
+    LadderLineBuilder() {
+        this.directions = new ArrayList<>();
+    }
+
+    public Manuel manuel() {
+        return new Manuel();
+    }
+
+    public Auto auto(DirectionCreateStrategy directionCreateStrategy) {
+        return new Auto(directionCreateStrategy);
+    }
+
+    private LadderLine finalBuild() {
+        return IntStream.range(0, directions.size())
+                .mapToObj(index ->
+                        new LadderPoint(index, directions.get(index))
+                )
+                .collect(
+                        Collectors.collectingAndThen(
+                                Collectors.toList(), LadderLine::new
+                        )
+                );
+    }
+
+    public class Manuel {
+        private Manuel() {}
+
+        public Manuel next(Direction direction) {
+            directions.add(direction);
+
+            return this;
+        }
+
+        public LadderLine build() {
+            return finalBuild();
+        }
+    }
+
+    public class Auto {
```

Comment:
```suggestion
    public static class Auto {
```

일반적으로 inner class의 경우 static 키워드를 작성해주는 것을 권장하는데요
구현하신 부분을 보면 static을 붙여주지 않은 것이 의도된 것 같아 혹시 static을 사용하지 않았을 때 문제를 알지만 이렇게 구현하신건지 궁금합니다 :)

Reply:
static을 붙이지 않았을떄는 메모리 관련 문제가 생길수 있다는건 알고 있지만
static을 붙였을때와 안 붙였을때의 작동 방식은 이해가 안되는것중 하나 입니다 😭
Auto 클래스에 인스턴스 변수가 있어 static을 붙이지 않았습니다..